bin/pelagos.py

```diff
@@ -149,16 +149,40 @@ def provision_node():
     app.logger.info("Data for provision: [" + str(request.get_data(as_text=True)) + "]")
     os = request.form['os']
     node_id = request.form['node']
+    add_sls = ''
     app.logger.info(
         'Provision node [{}]  with OS [{}]'.format(node_id, os))
+
+    if 'add_sls' in request.form:
+        add_sls = request.form['add_sls']
+        app.logger.info(
+            'Additinal salt script[{}]'.format(add_sls))
+
     node = _check_input_node(node_id)
     os_id = _check_input_os(os)
-    if app.simulate_mode == 'fast':
-        pxelinux_cfg.provision_node_simulate_fast(node,os)
-    elif app.simulate_mode == 'medium':
-        pxelinux_cfg.provision_node_simulate(node, os)
-    else:
-        pxelinux_cfg.provision_node(node, os)
+    try:
+        if app.simulate_mode == 'fast':
+            pxelinux_cfg.provision_node_simulate_fast(node,os)
+        elif app.simulate_mode == 'medium':
+            pxelinux_cfg.provision_node_simulate(node, os)
+        else:
+            pxelinux_cfg.provision_node(node, os, add_sls=add_sls.split())
+    except hw_node.TimeoutException as tmt_excp:
+        msg_tmt_excp= 'Caught TimeoutException %s' % tmt_excp
+        app.logger.info(msg_tmt_excp)
+        abort(504, msg_tmt_excp)
+    except hw_node.CannotBootException as boot_excp:
+        msg_boot_excp = 'Caught CannotBootException %s' % boot_excp
+        app.logger.info(msg_boot_excp)
+        abort(502, msg_boot_excp)
+    except hw_node.BMCException as bmc_excp:
+        msg_bmc_excp = 'Caught BMCException %s' % bmc_excp
+        app.logger.info(msg_bmc_excp)
+        abort(502, msg_bmc_excp)
```

Comment:
Is this the 502 code same as for CannotBootException set by intention?

Reply:
yes, in both cases it is from one source - node boot process, it could be bmc or ipmi or node issue. Also, I'm limited by a number of supported codes by flask engine.